internal/gatewayapi/contexts.go

```diff
@@ -516,6 +599,25 @@ func (r *RouteParentContext) SetCondition(route RouteContext, conditionType v1be
 		} else {
 			r.tlsRoute.Status.Parents[r.routeParentStatusIdx].Conditions = append(r.tlsRoute.Status.Parents[r.routeParentStatusIdx].Conditions, cond)
 		}
+	case KindTCPRoute:
+		for i, existing := range r.tcpRoute.Status.Parents[r.routeParentStatusIdx].Conditions {
+			if existing.Type == cond.Type {
```

Comment:
this seems a little hard to understand

Reply:
Agree, need to do some abstractions to this, but this need to refactor other Routes too.